imjoy/plugin.py

```diff
@@ -261,25 +266,43 @@ def launch_plugin(
 
         if envs is not None and len(envs) > 0:
             for env in envs:
-                print("Running env command: " + env)
-                logger.info("running env command: %s", env)
-                if env not in cmd_history:
-                    logging_callback("running env command: {}".format(env))
-                    code, errors = run_process(
-                        eng, pid, env.split(), env=plugin_env, cwd=work_dir
-                    )
+                if type(env) is str:
```

Comment:
Can't we add validation on the browser side so we always get a dict in the engine to parse?

Reply:
Yes, we can do that, let's do a dedicated PR for this sometime soon.

Reply:
It would be good to get that framework in place before we add features that make the backend more complex.